File: frontend/web/src/api/listings/getListing.ts
import axiosInstance from "../../utils/AxiosInstance.ts"

const getListing = async (id?: string) => {
    const token = localStorage.getItem("token");

    const { data } = await axiosInstance.get(`/api/listings/${id}`, {
        headers: {
            Authorization: `Bearer ${token}`,
        }
    });

    return data.listing;
}

export default getListing;

File: frontend/desktop/python-client/PyQt5App/api/listings/getListing.py
import requests
from PyQt5App.utils.Address import address
from PyQt5App.utils.sessionManager import SessionManager

class GetListing:
    def __init__(self, listing_id):
        self.address = address()
        self.sessionManager = SessionManager()
        self.token = self.sessionManager.get_token()
        self.listing_id = listing_id

    def getListing(self):
        response = requests.get(
            f"{self.address}/api/listings/{self.listing_id}",
            headers={
                "Authorization": f"{self.token}",
            }
        )
        data = response.json()
        return data.get("listing")